byteosaurus_hex.py

```diff
@@ -1933,7 +1942,388 @@ def vxlan():
+        # Get input parameters
+        mpls_input_param, common_param = requires("MPLS")
+        icmp_input_param, _ = requires('ICMP') 
+        inner_icmp_pkt, mpls_icmp_pkt = None, None
+        fuzzy = (input("Generate random MPLS ICMP Packet? (y/n) > ").strip()).lower()
+        if fuzzy == 'y':
+            # Build the inner ICMP packet
+            icmp_type = (input("ICMP Type (req/reply) > ").strip()).lower()
+            inputs = []
+            # Common parameters
+            for i in range(0, len(common_param)):
+                inputs.insert(i, input("{} > ".format(common_param[i])))
+            inner_icmp_pkt = icmp_packet(fuzzy, 'MPLS', icmp_type, inputs)
+            # Generate the Mpls packet
+            mpls_icmp_pkt = mpls_packet(fuzzy, inner_icmp_pkt, inputs) 
+            if inner_icmp_pkt != None and mpls_icmp_pkt != None:
+                logger.info("Inner ICMP packet built")
+                logger.info("MPLS ICMP Packet built")
+                mpls_icmp_pkt.show() 
+                return mpls_icmp_pkt, inputs[0], inputs[1]
+            else:
+                return None  
+        elif fuzzy == 'n':
+            labels = None
+            icmp_type = (input("ICMP Type (req/reply) > ").strip()).lower()
+            # Getting input parameters
+            inputs = []
+            del icmp_input_param[-1]  # Skipping VLAN tag for inner ICMP
+            for i in range(0, len(icmp_input_param)):
+                temp_input = input("Inner {} > ".format(icmp_input_param[i]))
+                inputs.insert(i, temp_input)
+            inner_icmp_pkt = icmp_packet(fuzzy, 'MPLS', icmp_type, inputs)
+            # Craft the MPLS packet. Get mpls input parameters
+            mpls_inputs = []
+            for i in range(0, len(mpls_input_param)):
+                temp_input = input("{} > ".format(mpls_input_param[i]))
+                if "Labels" in mpls_input_param[i]:
+                    #using a list comprehension with dummy variable x to create a list from the labels (which was a string)
+                    labels = [x.strip() for x in temp_input.split(',')]
+                    for k in range(0, len(labels)):
+                        try:
+                            if (0 <= int(labels[k]) <= 1048575) is True: #max MPLS label = 2^20
+                                continue
+                        except ValueError:
+                            logger.error(
+                                "Invalid input '{}' Expected comma separated values in range (0-1048575)".format(labels[k]))
+                            return None                  
+                    mpls_inputs.insert(i,labels)
+                elif "Codeword" in mpls_input_param[i] and temp_input.lower() == "y":
+                    mpls_inputs.insert(i, True)
+                elif "Codeword" in mpls_input_param[i] and temp_input.lower() == "n":
+                    mpls_inputs.insert(i, False)
+                elif "Codeword" not in mpls_input_param[i]:
```

Comment:
I'm wondering if this code section will ever reach / hit, because Codeword will always be part of mpls_input_params

Reply:
As discussed offline - `mpls_input_params[i]` could be either `Codeword` or one of the other 3 required inputs for MPLS. The `elif "Codeword" not in mpls_input_param[i]:` is the safety net that allows outer source MAC / destination MAC to be added to `mpls_inputs` (`Labels` also has a conditional `if "Labels" in mpls_input_param[i]`)